### PR TITLE
#411: Ignore unknown entries in go.mod

### LINKED ISSUE #411: Ignore unknown entries in go.mod
Project-keeper fails when parsing a `go.mod` file with any of the following entries:
* a commented line, starting with `//`
* a line like `replace github.com/docker/docker => github.com/docker/docker v20.10.3-0.20221013203545-33ab36d6b304+incompatible // 22.06 branch`
* a block like
```go
replace (
    github.com/docker/docker => github.com/docker/docker v20.10.3-0.20221013203545-33ab36d6b304+incompatible // 22.06 branch
)
```

Project-keeper must ignore any of these entries

### PR DESCRIPTION
**Pull request by kaklakariada**
Closes #411

### CODE REVIEW

File: project-keeper/src/main/java/com/exasol/projectkeeper/sources/analyze/golang/GoModFileParser.java [hunk 1]
```diff
@@ -12,23 +12,36 @@
...
-                    throw new IllegalStateException(ExaError.messageBuilder("E-PK-CORE-138")
-                            .message("Found unexpected line {{line}} in go.mod file", match.getLine()).toString());
+                RegexAction.create("^\\)$", match -> {
+                    this.insideRequireBlock = false;
+                    this.insideReplaceBlock = false;
+                }), RegexAction.create("^(?:require\\s+)?([^\\s]+)\\s+([^\\s/]+)(?:\\s*//\\s*(.*))?$", match -> {
+                    if (!insideReplaceBlock) {
+                        this.dependencies.add(createDependency(match.getGroups()));
+                    }
+                }), RegexAction.create(".*", match -> {
```
Comment (redcatbear) [inline project-keeper/src/main/java/com/exasol/projectkeeper/sources/analyze/golang/GoModFileParser.java:40]:
```suggestion
                }), //
                RegexAction.create("^(?:require\\s+)?([^\\s]+)\\s+([^\\s/]+)(?:\\s*//\\s*(.*))?$", match -> {
                    if (!insideReplaceBlock) {
                        this.dependencies.add(createDependency(match.getGroups()));
                    }
                }), //
                 RegexAction.create(".*", match -> {
```

File: project-keeper/src/test/java/com/exasol/projectkeeper/sources/analyze/golang/GoModFileTest.java [hunk 2]
```diff
@@ -20,6 +22,24 @@ void parseEmptyFile() {
...
+            "'// Trailing space '",
+            // Replace directives
+            "replace something with something", "replace\twith tab",
+            "replace github.com/docker/docker => github.com/docker/docker v20.10.3-0 // 22.06 branch",
+            // Replace block
+            "'replace (\n" //
+                    + "\tgithub.com/docker/docker => github.com/docker/docker v20.10.3-0.20221013203545-33ab36d6b304+incompatible // 22.06 branch\n"
+                    + "more content\n" //
+                    + ")'", })
+    void parseIgnoresLine(final String line) {
```
Comment (redcatbear) [inline project-keeper/src/test/java/com/exasol/projectkeeper/sources/analyze/golang/GoModFileTest.java:38]: Test methods should start with `test`

**Review by redcatbear — APPROVED** [review]

File: project-keeper/src/test/java/com/exasol/projectkeeper/sources/analyze/golang/GoModFileTest.java (hunk 2 above)
Comment (kaklakariada) [reply project-keeper/src/test/java/com/exasol/projectkeeper/sources/analyze/golang/GoModFileTest.java]: ✅

File: project-keeper/src/main/java/com/exasol/projectkeeper/sources/analyze/golang/GoModFileParser.java (hunk 1 above)
Comment (kaklakariada) [reply project-keeper/src/main/java/com/exasol/projectkeeper/sources/analyze/golang/GoModFileParser.java]: ✅

**Review by redcatbear — APPROVED** [review]

**Review by redcatbear — APPROVED** [review]

**Pull request merged**